src/lib/pipe_lib/strict_pipe.ml

```diff
@@ -121,51 +131,58 @@ module Reader0 = struct
       don't_wait_for
         (let%map () = Deferred.List.iter readers ~f:Pipe.closed in
          Pipe.close_read reader.reader) ;
+      reader.downstreams <- wrap_downstreams readers;
```

Comment:
You are creating two disjoint sets of `Reader.t's`: one which is assigned to `downstream`, and one which is returned by this function. You should just call `List.map readers ~f:wrap_reader`, assign the value of that into `downstreams`, and then return those.

Reply:
The requested changes have been done. Can you help me review it?